src/widgetastic/browser.py

```diff
@@ -243,7 +243,9 @@ def elements(
                 if isinstance(parent, Browser):
                     root_element = parent.selenium
                 elif isinstance(parent, (WebElement)) or hasattr(parent, '__locator__'):
-                    root_element = self.element(parent, check_visibility=check_visibility)
+                    grandparent = getattr(parent, 'parent', None)
+                    root_element = self.element(parent, parent=grandparent,
+                                                check_visibility=check_visibility)
```

Comment:
I believe situations like this are (should be) handled in the browser wrapper. I don't think skipping "generations" is entirely correct.